### DISCUSSION TITLE
[Feature Request] upgrade legacy to alert rules and export

### DISCUSSION THREAD

**Discussion by jgournet** [grafana/grafana#72178] [Alerting]
This is a bit far fetched, but that's something that would GREATLY help adoption of alerting rule.

Flow would be something like:
- enable LEGACY ALERTING in grafana 9.
- have a new endpoint in grafana 9 that could be called something like /export-upgrade/ALERT_ID   <= this would output a JSON for the converted legacy alert to alert rule.

I'm asking this because:
- the "export" functionality already exists in grafana 9 and 10.
- the "upgrade" functionality already exists when upgrading from grafana 8 to 9 (as far as I know?)
So hopefully, this could be easy enough to implement.

This would help all the people like me who use alert/alert rules provisioning and ephemeral grafana pod/docker, as we cannot just "upgrade" those.

**Comment by jgournet** [comment]
Alright ... found "a way" to do this - it's a bit painful, but might help people:

1) Prepare a quick image:
```
FROM ubuntu:latest

RUN apt-get update

RUN apt-get install -y wget adduser libfontconfig1
WORKDIR /mnt
```

2) Run it and install Grafana 8 with legacy alerting:
```
docker build -t test .
docker run -it --network host -p 3000:3000 -v `pwd`:/mnt test
export GF_ALERTING_ENABLED=true
export GF_UNIFIED_ALERTING_ENABLED=false
export GF_AUTH_ANONYMOUS_ENABLED=true

dpkg -i grafana-enterprise_8.5.11_amd64.deb

# optional: add datasources:
#cp datasource.yaml /etc/grafana/provisioning/datasources/

service grafana-server start
```

3) Manually import dashboards with alerts:
- navigate to localhost:3000
- go to dashboard
- go to "import"
- import 1 dashboard
- once imported: click on "save" icon !
- add more dashboards

4) Upgrade grafana:
```
service grafana-server stop
unset GF_ALERTING_ENABLED
unset GF_UNIFIED_ALERTING_ENABLED
dpkg -i grafana-enterprise_9.4.7_amd64.deb
service grafana-server start
```

5) Export alerts rules:
```
curl -s -u admin:admin http://localhost:3000/api/v1/provisioning/alert-rules/export?format=json | jq ''
```

6) clean them up manually

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82541. Feel free to continue the discussion around this there. Thank you!
